src/Metadata/DateTimeOptions.php

```diff
@@ -20,17 +20,18 @@ final class DateTimeOptions implements \JsonSerializable
     private $zone;
 
     /**
-     * Use if a different format should be used for parsing dates than for generating dates.
+     * Use if different formats should be used for parsing dates than for generating dates.
      *
-     * @var string|null
+     * @var string[]|null
      */
-    private $deserializeFormat;
+    private $deserializeFormats;
 
-    public function __construct(?string $format, ?string $zone, ?string $deserializeFormat)
+    public function __construct(?string $format, ?string $zone, ?string $deserializeFormat, ?array $allDeserializeFormats = [])
     {
         $this->format = $format;
         $this->zone = $zone;
-        $this->deserializeFormat = $deserializeFormat;
+        $deserializeFormat = \is_string($deserializeFormat) ? [$deserializeFormat] : $deserializeFormat;
+        $this->deserializeFormats = $allDeserializeFormats ?: $deserializeFormat;
```

Comment:
this means that if both are specified, the $deserializeFormat *must* be included in $allDeserializeFormats?

Reply:
in the interest of a seamless upgrade path, i suggest we instead remove the type restriction and phpdoc say it can be `null|string|string[]` and keep just one parameter. then implementations that upgrade to the 1.x version that has this change can start to pass the array already, and when we move to 2.0, we no longer allow string and add array as type declaration.

if we have the single parameter and remove that in version 2, it makes for a more complicated change as a using code can not easily support 1.x and 2.x